### DISCUSSION TITLE
Section Page Headers

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [QuestPDF/QuestPDF#297] [Q&A] [answered]
Hi all,

I have a page with several sections. At the beginning of each new section, I want to show the page header again. To facilitate this, I have created an instance of `IComponent` called `SectionHeader`, but I'm unsure how to ensure that this only shows on page one of each section. `IContainer` has `ShowIf()` that I could in theory leverage, but I'm not sure how to write that check.

An alternative here is to create a new page instead of a new section and use `ShowOnce()` in the header to make it only show up on the first page. Each section should start on its own page anyways, so this would presumably help facilitate that.

Thanks!

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The approach you described — using separate pages with `ShowOnce` in the header — is the recommended solution. A dedicated example for this pattern is now available in the documentation:
https://www.questpdf.com/examples/custom-header-on-first-page.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
